Composer/packages/client/src/store/action/project.ts

```diff
@@ -12,6 +15,25 @@ import settingStorage from './../../utils/dialogSettingStorage';
 import luFileStatusStorage from './../../utils/luFileStatusStorage';
 import httpClient from './../../utils/httpUtil';
 
+export const checkProjectUpdates = async () => {
+  return new Promise(resolve => {
+    const timer = setInterval(() => {
+      //before open a new bot, we should check the task queue, lu/lg worker
+      if (filePersistence.isEmpty() && lgWorker.isEmpty() && luWorker.isEmpty()) {
```

Comment:
Is there a way to tell the workers to drain their queues instead of asking if they are empty?

I imagine something like:

```
allWork = []
for worker in [filePersistence, lgWorker, luWorker]
  allWork.push(worker.flush());

await Promise.all(allWork);
```